### PR TITLE
Add RA location projections

### PR DESCRIPTION
**Pull request by arothuis**
[116602125](https://www.pivotaltracker.com/story/show/116602125)

### CODE REVIEW

File: src/Surfnet/StepupMiddleware/ApiBundle/Configuration/Projector/RaLocationProjector.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+namespace Surfnet\StepupMiddleware\ApiBundle\Configuration\Projector;
+
+use Broadway\ReadModel\Projector;
+use Surfnet\Stepup\Configuration\Event\RaLocationAddedEvent;
+use Surfnet\StepupMiddleware\ApiBundle\Configuration\Entity\RaLocation;
+use Surfnet\StepupMiddleware\ApiBundle\Configuration\Repository\RaLocationRepository;
+
+class RaLocationProjector extends Projector
+{
+    private $repository;
```
Comment (rjkip) [inline src/Surfnet/StepupMiddleware/ApiBundle/Configuration/Projector/RaLocationProjector.php]: :goberserk:

File: src/Surfnet/StepupMiddleware/ApiBundle/Doctrine/Type/ConfigurationContactInformationType.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+namespace Surfnet\StepupMiddleware\ApiBundle\Doctrine\Type;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Types\ConversionException;
+use Doctrine\DBAL\Types\Type;
+use Surfnet\Stepup\Exception\InvalidArgumentException;
+use Surfnet\Stepup\Configuration\Value\ContactInformation;
+
+/**
+ * Custom Type for the Location Value Object for the Configuration domain
```
Comment (rjkip) [inline src/Surfnet/StepupMiddleware/ApiBundle/Doctrine/Type/ConfigurationContactInformationType.php]: > Location Value Object

File: src/Surfnet/StepupMiddleware/ApiBundle/Doctrine/Type/ConfigurationContactInformationType.php [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        return $platform->getClobTypeDeclarationSQL($fieldDeclaration);
+    }
+
+    public function convertToDatabaseValue($value, AbstractPlatform $platform)
+    {
+        if (is_null($value)) {
+            return $value;
+        }
+
+        return (string) $value;
```
Comment (rjkip) [inline src/Surfnet/StepupMiddleware/ApiBundle/Doctrine/Type/ConfigurationContactInformationType.php]: :goberserk:

File: src/Surfnet/StepupMiddleware/ApiBundle/Doctrine/Type/InstitutionType.php [hunk 4]
```diff
@@ -49,7 +51,20 @@ public function convertToPHPValue($value, AbstractPlatform $platform)
             return $value;
         }
 
-        return new Institution($value);
+        try {
+            $institution = new Institution($value);
+        } catch (InvalidArgumentException $e) {
+            // get nice standard message, so we can throw it keeping the exception chain
+            $doctrineExceptionMessage = ConversionException::conversionFailedFormat(
```
Comment (rjkip) [inline src/Surfnet/StepupMiddleware/ApiBundle/Doctrine/Type/InstitutionType.php]: Thank you!

**Comment by rjkip** [comment]
Lookin' good! Just the projector service left I guess?

File: app/DoctrineMigrations/Version20160617153313.php [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+{
+    /**
+     * @param Schema $schema
+     */
+    public function up(Schema $schema)
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() != 'mysql', 'Migration can only be executed safely on \'mysql\'.');
+
+        $this->addSql('CREATE TABLE ra_location (ra_location_id VARCHAR(36) NOT NULL, institution LONGTEXT NOT NULL, ra_location_name VARCHAR(255) NOT NULL, location LONGTEXT NOT NULL, contact_information LONGTEXT NOT NULL, INDEX idx_ra_location_institution (institution), PRIMARY KEY(ra_location_id)) DEFAULT CHARACTER SET utf8 COLLATE utf8_unicode_ci ENGINE = InnoDB');
```
Comment (arothuis) [inline app/DoctrineMigrations/Version20160617153313.php]: Hmm. This seems to fail:
```
SQLSTATE[42000]: Syntax error or access violation: 1170 BLOB/TEXT column 'institution' used in key specification without a key
```

**Comment by arothuis** [comment]
@rjkip

**Comment by rjkip** [comment]
:ship::ship::ship:

File: src/Surfnet/StepupMiddleware/MiddlewareBundle/Console/Command/MigrationsDiffDoctrineCommand.php [hunk 6]
```diff
@@ -45,12 +45,24 @@ public function execute(InputInterface $input, OutputInterface $output)
                 $output->write($data);
             });
 
+        // Quick migrations will cause Gateway migrations to overwrite Middleware's as they are identified per second
+        sleep(1);
```
Comment (relaxnow) [inline src/Surfnet/StepupMiddleware/MiddlewareBundle/Console/Command/MigrationsDiffDoctrineCommand.php]: Snorlax used Rest. It's super effective!

**Pull request merged**